### ISSUE TITLE
Add support for LOAD INTO statement, BigQuery dialect

### ISSUE THREAD

**Issue by VMois** [sqlfluff/sqlfluff#6589]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

I am using version `sqlfluff, version 3.1.0`, and [LOAD INTO statement](https://cloud.google.com/bigquery/docs/reference/standard-sql/load-statements#load_data_statement) is not supported in BigQuery dialect.

Example:
```sql
LOAD DATA INTO mydataset.table1
  FROM FILES(
    format='AVRO',
    uris = ['gs://bucket/path/file.avro']
  )

```

Tool output:
```bash
$ sqlfluff lint test.sql --dialect bigquery
== [test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'LOAD DATA
                       | INTO mydataset.table1\n  FROM F...'
L:   6 | P:   1 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

Happy to work on MR. I would appreciate any pointers on what needs to be added (what files modified) to add LOAD INTO. Thank you.

### Use case

When an SQL file contains a LOAD INTO statement, it should be parsable to properly lint the statement and other parts of the SQL.

You can find a situation when the LOAD INTO statement is used with SELECT and others to filter data. So the whole file should be parsable.

### Dialect

BigQuery dialect.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Thanks for wanting to contribute! You'll want to add the change in [dialect_bigquery.py](https://github.com/sqlfluff/sqlfluff/blob/96bafc1a9425d2093d46f29c0c3f0cde826e8487/src/sqlfluff/dialects/dialect_bigquery.py). I'd recommend drawing inspiration from [Snowflake's Copy Into Statement](https://github.com/sqlfluff/sqlfluff/blob/96bafc1a9425d2093d46f29c0c3f0cde826e8487/src/sqlfluff/dialects/dialect_snowflake.py#L6080).

Also you'll want to check out the [CONTRIBUTING.md](https://github.com/sqlfluff/sqlfluff/blob/96bafc1a9425d2093d46f29c0c3f0cde826e8487/CONTRIBUTING.md)

Feel free to ask if you have any questions!
